src/frequenz/sdk/_internal/_channels.py

```diff
@@ -92,6 +92,10 @@ async def stop(self) -> None:
         """Stop the cache."""
         await cancel_and_await(self._task)
 
+    def __del__(self) -> None:
+        """Stop the cache when it is deleted."""
+        self._task.cancel()
```

Comment:
Best to check if the event loop is still running before calling cancel, else, we might get more warning spam in tests.

Reply:
Oh, really? Actually I added this because I was getting warnings in tests about the last value cache tasks being pending when the loop was destroyed. But I debugged the issue and the tasks were properly finished, it seems it might be a bug in pytest-async, I believe we had some similar issues in the past.

Maybe I should remove this and that's it, I think according to Python specs this should not be necessary, and tasks should be cancelled (or discarded) when the even loop is closed, and dangling tasks will also be removed from the loop and finished, it might only happen that this is delayed until the GC runs if the reference was part of a reference loop.

I'd say until we are completely sure this is really necessary we just don't add it.

Reply:
Just to add more context, this is the issue that made me try this:

* https://github.com/frequenz-floss/frequenz-sdk-python/pull/962#issuecomment-2182264984